sympy/algebras/quaternion.py

```diff
@@ -421,22 +420,23 @@ def pow(self, p):
         >>> q.pow(4)
         668 + (-224)*i + (-336)*j + (-448)*k
         """
+        p = sympify(p)
         q = self
         if p == -1:
             return q.inverse()
         res = 1
 
+        if fuzzy_not(p.is_Integer):
+            return NotImplemented
```

Comment:
I think `p.is_Integer` is equivalent to `isinstance(p, Integer)` so is always (non-symbolic) True or False. This should be generally true of any `is_Capitalised` attribute but this is not always done consistently. I don't think that the `fuzzy_not` is needed here.

Reply:
Yes. is_integer could be None but not is_Integer

Reply:
OK, will fix!